pkg/server/sandbox_status.go

```diff
@@ -82,62 +86,70 @@ func (c *criContainerdService) PodSandboxStatus(ctx context.Context, r *runtime.
 
 // TODO (mikebrow): discuss predefining constants structures for some or all of these field names in CRI
 type sandboxInfo struct {
-	Pid         uint32            `json:"pid"`
-	State       string            `json:"state"`
-	SnapshotKey string            `json:"snapshotKey"`
-	Snapshotter string            `json:"snapshotter"`
-	RuntimeSpec *runtimespec.Spec `json:"runtimeSpec"`
+	Pid         uint32                    `json:"pid"`
+	Status      string                    `json:"processStatus"`
+	NetNS       string                    `json:"netNamespace"`
+	NetNSClosed bool                      `json:"netNamespaceClosed"`
+	Image       string                    `json:"image"`
+	SnapshotKey string                    `json:"snapshotKey"`
+	Snapshotter string                    `json:"snapshotter"`
+	Config      *runtime.PodSandboxConfig `json:"config"`
+	RuntimeSpec *runtimespec.Spec         `json:"runtimeSpec"`
 }
 
-// toCRISandboxContainerInfo converts internal container object information to CRI sandbox container status response info map.
-func toCRISandboxContainerInfo(ctx context.Context, container containerd.Container, verbose bool) (map[string]string, error) {
+// toCRISandboxInfo converts internal container object information to CRI sandbox status response info map.
+func toCRISandboxInfo(ctx context.Context, sandbox sandboxstore.Sandbox,
+	pid uint32, processStatus containerd.ProcessStatus, verbose bool) (map[string]string, error) {
 	if !verbose {
 		return nil, nil
 	}
 
-	task, err := container.Task(ctx, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get sandbox container task: %v", err)
-	}
-	status, err := task.Status(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get sandbox container task status: %v", err)
+	si := &sandboxInfo{
+		Pid:    pid,
+		Status: string(processStatus),
+		NetNS:  sandbox.NetNSPath,
+		Config: sandbox.Config,
 	}
 
-	info := make(map[string]string)
-	pid := task.Pid()
+	if si.Status == "" {
+		// If processStatus is empty, it means that the task is deleted. Apply "deleted"
+		// status which does not exist in containerd.
+		si.Status = "deleted"
+	}
 
-	oldSpec, err := container.Spec(ctx)
-	if err != nil {
-		glog.Errorf("Failed to get sandbox container runtime spec: %v", err)
+	if sandbox.NetNSPath != "" {
+		// Add network closed information if sandbox is not using host network.
+		si.NetNSClosed = (sandbox.NetNS == nil || sandbox.NetNS.Closed())
 	}
 
-	var snapshotkey, snapshotter string
-	ctrInfo, err := container.Info(ctx)
+	container := sandbox.Container
+
+	spec, err := container.Spec(ctx)
 	if err == nil {
-		snapshotkey = ctrInfo.SnapshotKey
-		snapshotter = ctrInfo.Snapshotter
+		si.RuntimeSpec = spec
 	} else {
-		glog.Errorf("Failed to get target shapshot info: %v", err)
```

Comment:
preferred it this way

Reply:
Now we also get sandbox container image, it's not only snapshot info.

Reply:
The call to container.Spec and container.Info and error checking are in both versions just moved around in this one. The prior way used variables for temp storage in the function then did a "to" sandboxInfo at the end. The new one creates the sandboxInfo early and has various assigns as it goes. A lot of style changes mixed in with the actual changes, makes it hard to see what's new vs. what is just changed for style sake.